src/components/recs/rec-tile.tsx

```diff
@@ -1,53 +1,282 @@
+import { Button } from "../ui/button";
+import { toast } from "../ui/use-toast";
+import { Dialog, DialogTrigger, DialogContent, DialogHeader, DialogFooter, DialogTitle, DialogDescription } from "../ui/dialog";
+import { Filters } from "@/types/Filters";
 
 interface RecTileProps {
+  id: string;
   rec: IRecordedGame;
   showMap?: boolean;
+  refetchRecs: (filters: Filters) => void;
+  filters: Filters;
 }
 
-export default function RecTile({ rec, showMap = true }: RecTileProps) {
+export default function RecTile({ id, rec, showMap = true,refetchRecs, filters }: RecTileProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [fileName, setFileName] = useState(rec.gameTitle);
+  const [dialogOpen, setDialogOpen] = useState(false);
   const windowSize = useContext(WindowContext);
   const { leftTeams, rightTeams } = useTeams(rec);
+  const [isProcessing, setIsProcessing] = useState(false);
+  const recGameAuthor = id === rec.uploadedByUserId;
+
+  async function handleEditGameTitle(fileName: string, gameGuid: string) {
+    setIsProcessing(true);
+    const response = await fetch(`/api/recordedGames`, {
+      method: "PUT",
+      headers: { "content-type": "application/json" },
+      body: JSON.stringify({ gameTitle: fileName, gameGuid, userId: id }),
+    });
+    if (response.ok) {
+      toast({
+        title: "Game Title updated successfully",
+        duration: 3000,
+      });
+      refetchRecs(filters);
+    } else if (response.status === 402) {
+      toast({
+        title: "Wrong user",
+        duration: 3000,
+      });
+    }
+    else {
+      toast({
+        title: "Error updating the title",
+        duration: 3000,
+      });
+    }
+    setIsProcessing(false);
+    setIsOpen(false);
+  }
 
+  async function handleDeleteGame(gameGuid: string) {
+    setIsProcessing(true);
+    const response = await fetch(`/api/recordedGames`, {
+      method: "DELETE",
+      headers: { "content-type": "application/json" },
+      body: JSON.stringify({ gameGuid, userId: id }),
+    });
+    if (response.ok) {
+      toast({
+        title: "Game deleted successfully",
+        duration: 3000,
+      });
+      refetchRecs(filters);
+    } else if (response.status === 402) {
+      toast({
+        title: "Wrong user",
+        duration: 3000,
+      });
+    } else {
+      toast({
+        title: "Error deleting the game",
+        duration: 3000,
+      });
+    }
+    setIsProcessing(false);
+    setIsOpen(false);
+    setDialogOpen(false);
+  }
   return (
     <div>
       {windowSize && windowSize.width >= 768 ? (
         // desktop layout
         <div>
           <div className="flex">
-            {leftTeams}
+            <div className="flex mt-9">{leftTeams}</div>
             <div>
               <RecTitle gameTitle={rec.gameTitle || ""} />
               {showMap && <RecMap rec={rec} />}
             </div>
-            {rightTeams}
+            <div className="flex flex-col">
```

Comment:
Is this all for the slideout "edit" component? I think it should be separated into its own component rather than defined in line here if you can pull this out.